lib/spack/spack/ci_optimization.py

```diff
@@ -0,0 +1,375 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import collections
+
+try:
+    import collections.abc as collections_abc
+except ImportError:
+    collections_abc = collections
```

Comment:
It seems very unlikely that vermin is going to fix the problem, since it would require a level of code analysis that probably can't be done statically.

I'll see if I can sidestep the issue in this case, but I think a more comprehensive solution is warranted.  To my knowledge, the `six` compatibility package doesn't include an import for the `collections.abc` module, and so the vermin command as-is, combined with our need to continue supporting Python 2 and 3, effectively precludes any use of the collections abstract base classes.